Wrappers/Python/cil/processors/Masker.py

```diff
@@ -0,0 +1,168 @@
+# -*- coding: utf-8 -*-
+#   This work is part of the Core Imaging Library (CIL) developed by CCPi 
+#   (Collaborative Computational Project in Tomographic Imaging), with 
+#   substantial contributions by UKRI-STFC and University of Manchester.
+
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+
+#   http://www.apache.org/licenses/LICENSE-2.0
+
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+from cil.framework import DataProcessor, AcquisitionData, ImageData, ImageGeometry, DataContainer
+import warnings
+import numpy
+
+class Masker(DataProcessor):
+    r'''
+    Processor to fill missing values provided by mask. Please use the desiried method to configure a processor for your needs.
+    '''
+
+    @staticmethod
+    def value(mask=None, value=0):
+        r'''This imputes value where mask == 0.
+        :param mask: mask 
+        :type mask: DataContainer
+        :param value: values to be assigned to missing elements
+        :type value: float, default=0
+        '''
+
+        processor = Masker(mode='value', mask=mask, value=value)
+
+        return processor
+    
+    @staticmethod
+    def mean(mask=None, axis=None):
+        r'''This imputes mean where mask == 0. If no axis is specified then operates over flattened array.
+        :param mask: mask 
+        :type mask: DataContainer
+        :param axis: specify axis as int or from 'dimension_labels' to calculate mean. 
+        :type axis: str, int
+        '''
+
+        processor = Masker(mode='mean', mask=mask, axis=axis)
+
+        return processor
+    
+    @staticmethod
+    def median(mask=None, axis=None):
+        r'''This imputes median where mask == 0. If no axis is specified then operates over flattened array.
+        :param mask: mask 
+        :type mask: DataContainer
+        :param axis: specify axis as int or from 'dimension_labels' to calculate median. 
+        :type axis: str, int
+        '''
+
+        processor = Masker(mode='median', mask=mask, axis=axis)
+
+        return processor
+
+    def __init__(self,
+                 mask = None,
+                 mode = 'value',
+                 value = 0,
+                 axis = None ):
+        
+        r'''Processor to fill missing values provided by mask.
+        :param mask: DataContainer containing a boolean array with zeros where outliers are detected
```

Comment:
:param mask: A boolean array with the same dimensions as input, where 'False' represents masked values. Mask can be generated using 'MaskGenerator' processor to identify outliers.
:type mask: DataContainer, ImageData, AcquisitionData, numpy.ndarray,